slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c

```diff
@@ -2340,6 +2340,59 @@ int getClockDivider(enum CLKINDEX ind) {
     return clkDivider[ind];
 }
 
+int setBadChannels(int nch, int *channels) {
+    LOG(logINFO, ("Setting %d bad channels\n", nch));
+    memset(badChannelMask, 0, NCHAN_PER_MODULE * sizeof(char));
+
+    // setting badchannels, loop through list
+    for (int i = 0; i < nch; ++i) {
+        LOG(logINFO, ("\t[%d]: %d\n", i, channels[i]));
+        for (int ich = channels[i] * NCOUNTERS;
+             ich != channels[i] * NCOUNTERS + 3; ++ich) {
+            badChannelMask[ich] = 1;
+        }
+    }
+
+    for (int i = 0; i != NCHAN_PER_MODULE; ++i) {
+        if (badChannelMask[i]) {
+            LOG(logDEBUG1, ("[%d]:0x%02x\n", i, badChannelMask[i]));
+        }
+    }
+    return setTrimbits(detectorChans);
+}
+
+int *getBadChannels(int *nch) {
+    int *retvals = NULL;
+    // number of bad channels
+    *nch = 0;
+    for (int ich = 0; ich != NCHAN_PER_MODULE; ++ich) {
+        *nch += __builtin_popcount(badChannelMask[ich]);
```

Comment:
I think this is correct but since we moved away from int and setting bits maybe 

```
if(badChannelMask)
    *nch += 1
```

Better communicates the logic?

Reply:
Agree. Done